src/main/java/net/minecraftforge/fluids/BlockFluidClassic.java

```diff
@@ -368,6 +368,17 @@ public int place(World world, BlockPos pos, @Nonnull FluidStack fluidStack, bool
         }
         if (doPlace)
         {
+            if (!world.isRemote)
```

Comment:
This block of code is repeated three times - maybe move it to a method that can be called from all places?